### PR TITLE
removed subscription_id from azure_keyvault_secret lookup when using az cli auth

### LINKED ISSUE #1174: azure_keyvault_secret lookup plugin doesn't work without subscription_id with az cli authentication
##### SUMMARY

azure_keyvault_secret lookup plugin doesn't work without subscription_id with az cli authentication. I'm not fully sure why subscription ID is required when we are connecting via vault_url.
Yesterday I started working on preparing support for az cli authentication. Today I noticed that it was added by @Fred-sun 
I prepared change. In my version of the solution, I'm using `DefaultAzureCredential` method and subscription_id is not needed.
I'll open the PR and we can consider whether we want to simplify it a bit.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure_keyvault_secret

##### EXPECTED RESULTS

```
ok: [localhost] => {
    "msg": "secret-value"
}
```

##### ACTUAL RESULTS

```
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'azure_keyvault_secret'. Error was a <class 'UnboundLocalError'>, original message: local variable 'credentials' referenced before assignment. local variable 'credentials' referenced before assignment"}
```

### PR DESCRIPTION
**Pull request by msl0**
Added support for Azure CLI authentication without subscription ID in azure_keyvault_secret lookup plugin, fixes #1174

### CODE REVIEW

**Comment by msl0** [comment]
#1174

**Comment by msl0** [comment]
#1175

**Comment by Fred-sun** [comment]
@msl0 I recommend adding support for "az login" credentials, as there are users using this plugin even through az login! Thanks!

**Comment by msl0** [comment]
@Fred-sun support for "az login" is included. [`DefaultAzureCredential()`](https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) supports multiple authentication scenarios. It can use credentials from environment variables and works with managed identity, identity currently logged in to the Azure CLI and PowerShell

**Comment by Fred-sun** [comment]
@msl0 Ok, I'll test it out as soon as possible! Thanks!

File: plugins/lookup/azure_keyvault_secret.py [hunk 1]
```diff
@@ -32,21 +33,22 @@
...
         description: Tenant id of service principal.
-    subscription_id:
-        description: Your Azure subscription Id.
 notes:
     - If version is not provided, this plugin will return the latest version of the secret.
     - If ansible is running on Azure Virtual Machine with MSI enabled, client_id, secret and tenant isn't required.
     - For enabling MSI on Azure VM, please refer to this doc https://docs.microsoft.com/en-us/azure/active-directory/managed-service-identity/
     - After enabling MSI on Azure VM, remember to grant access of the Key Vault to the VM by adding a new Acess Policy in Azure Portal.
     - If MSI is not enabled on ansible host, it's required to provide a valid service principal which has access to the key vault.
+    - To authenticate via service principal, pass client_id, secret and tenant_id or set environment variables AZURE_CLIENT_ID, AZURE_CLIENT_SECRET and AZURE_TENANT_ID.
```
Comment (Fred-sun) [inline plugins/lookup/azure_keyvault_secret.py]: line too long (168 > 160 characters）

**Comment by Fred-sun** [comment]
@msl0 There is a conflict in the file you submitted. Can you fix it?

**Comment by msl0** [comment]
@Fred-sun fixed and then tested once again

**Comment by msl0** [comment]
@Fred-sun did you have a chance to take a look at my PR?

File: plugins/lookup/azure_keyvault_secret.py (hunk 1 above)
Comment (Fred-sun) [inline plugins/lookup/azure_keyvault_secret.py]: line too long (168 > 160 characters)

File: plugins/lookup/azure_keyvault_secret.py [hunk 2]
```diff
@@ -108,23 +110,29 @@
...
+    from azure.keyvault.secrets import SecretClient
+
 except ImportError:
     pass
 
 display = Display()
 
 TOKEN_ACQUIRED = False
 
+logger = logging.getLogger("azure.identity").setLevel(logging.ERROR)
```
Comment (Fred-sun) [inline plugins/lookup/azure_keyvault_secret.py:134]: name 'logging' is not defined

File: plugins/lookup/azure_keyvault_secret.py [hunk 3]
```diff
@@ -108,23 +110,29 @@
...
 try:
     import requests
     from azure.keyvault.secrets import SecretClient
     import logging
-    import os
-    from azure.common.credentials import get_cli_profile
-    from azure.identity._credentials.client_secret import ClientSecretCredential
+    from azure.common.exceptions import ClientRequestError
+    from msrest.exceptions import ClientRequestError
+    from azure.keyvault.models import KeyVaultErrorException
```
Comment (Fred-sun) [inline plugins/lookup/azure_keyvault_secret.py]: Useless definition
```suggestion

```

**Comment by Fred-sun** [comment]
@msl0 Being tested. Thank you!

File: plugins/lookup/azure_keyvault_secret.py (hunk 1 above)
Comment (msl0) [reply plugins/lookup/azure_keyvault_secret.py]: Fixed

File: plugins/lookup/azure_keyvault_secret.py (hunk 3 above)
Comment (msl0) [reply plugins/lookup/azure_keyvault_secret.py]: Thanks, suggestion appied

**Comment by Fred-sun** [comment]
@msl0 Have you tested the changed 'azure_keyvault_secret.py' locally? Can you get tocken and get the secret value? Thank you!

**Comment by Fred-sun** [comment]
@msl0 Everything else is fine, But there's one sanity test that always has a problem, you can't find the cause. Can you adjust the location of 'import logging'? Avoid this mistake. Please change line 121 below line 118. Thank you!
```e.g.
ERROR: plugins/lookup/azure_keyvault_secret.py:134:0: traceback: NameError: name 'logging' is not defined

**Comment by msl0** [comment]
@Fred-sun Thank you! I changed the import order if that's what you meant. But I don't fully understand the problem. I didn't notice anything when testing the change locally so maybe there is some problem with the test

**Review by xuzhang3 — APPROVED** [review]

**Pull request merged**